src/core/org/apache/jmeter/util/JSR223TestElement.java

```diff
@@ -86,11 +88,16 @@ public JSR223TestElement() {
     }
 
     protected ScriptEngine getScriptEngine() throws ScriptException {
-        final String lang = getScriptLanguage();
+        String lang = getScriptLanguage();
+        
+        if (lang.isEmpty()) {
+            lang = defaultScriptLanguage;
+            log.warn("Script language has not been chosen on the UI, the script will be interpreted as a groovy script");
```

Comment:
Do you really intend to warn on each and every sampler execution?
This looks awfully broken.